web/src/main/java/com/bazaarvoice/emodb/web/scanner/writer/DefaultScanWriterGenerator.java

```diff
@@ -0,0 +1,37 @@
+package com.bazaarvoice.emodb.web.scanner.writer;
+
+import com.bazaarvoice.emodb.web.scanner.ScanDestination;
+import com.google.common.base.Optional;
+import com.google.inject.Inject;
+import java.net.URI;
+
+public class DefaultScanWriterGenerator extends ScanWriterGenerator {
+    private final ScanWriterFactory _scanWriterFactory;
+
+    @Inject
+    public DefaultScanWriterGenerator(ScanWriterFactory scanWriterFactory) {
+        _scanWriterFactory = scanWriterFactory;
+    }
+
+    /**
+     * Creates a scan writer for the given destination.
+     */
+    public ScanWriter createScanWriter(int taskId, ScanDestination destination) {
+        if (destination.isDiscarding()) {
+            return _scanWriterFactory.createDiscardingScanWriter(taskId, Optional.<Integer>absent());
+        }
+
+        URI uri = destination.getUri();
+        String scheme = uri.getScheme();
+
+        if ("file".equals(scheme)) {
```

Comment:
nitpick, could be changed to `switch/case`

Reply:
This is actually the original code from `ScanWriterGenerator`, and I think it is best not to refactor it if not absolutely necessary.

Reply:
ok